### ISSUE TITLE
Port System.Net.WebClient missing members

### ISSUE THREAD

**Issue by ericeil** [dotnet/corefx#12137]
```csharp
    public class WebClient : Component {
        public RequestCachePolicy CachePolicy { get; set; }
        protected virtual void OnDownloadProgressChanged(DownloadProgressChangedEventArgs e);
        protected virtual void OnUploadProgressChanged(UploadProgressChangedEventArgs e);
    }
```

**Comment by ericeil** [comment]
Note that we are currently *not* deriving from `Component`.

**Comment by stephentoub** [comment]
> Note that we are currently not deriving from Component.

Once @AlexGhiondea completes adding Component to corefx, I suggest that someone do a sweep through the repo making Component a base class for everything it needs to be, rather than doing it piecemeal.
cc: @danmosemsft

**Comment by danmosemsft** [comment]
Opened #12154

**Comment by stephentoub** [comment]
#12151 fixed the missing virtuals, and #12154 coves deriving from Component, so all that remains is CachePolicy, which is in the source but commented out until corefx updates to the latest packages and WebClient can see the System.Net.Cache types added to System.Net.Requests.
